nemo/lightning/pytorch/callbacks/preemption.py

```diff
@@ -61,13 +64,15 @@ def on_train_end(self, trainer: Trainer, pl_module) -> None:
 
     def on_train_batch_end(self, trainer: Trainer, pl_module, outputs, batch, batch_idx: int) -> None:
         if self.interrupted:
-            logging.info("Preemption detected, signaling trainer to stop")
-            trainer.should_stop = True
-
-    def on_exception(self, trainer: Trainer, pl_module, exception: BaseException) -> None:
-        if isinstance(exception, PreemptionException):
-            logging.info("Handling PreemptionException")
+            logging.info("Preemption detected, saving checkpoint and exiting")
             trainer.should_stop = True
+            if any(map(lambda callback: isinstance(callback, ModelCheckpoint), trainer.callbacks)):
```

Comment:
I think you can just use the [checkpoint_callback](https://github.com/Lightning-AI/pytorch-lightning/blob/d25014dbda553eabe12fb0669e7451ad7aa0c576/src/lightning/pytorch/trainer/trainer.py#L1305) property here, right?